src/main/java/org/mockito/mock/MockCreationSettings.java

```diff
@@ -73,6 +73,16 @@
     boolean isUsingConstructor();
 
     /**
+     * Used when arguments should be passed to the mocked object's constructor, regardless of whether these
+     * arguments are supplied directly, or whether they include the outer instance.
+     *
+     * @return An array of arguments that are passed to the mocked object's constructor. If
+     * {@link #getOuterClassInstance()} is available, it is prepended to the passed arguments.
+     */
+    @Incubating
+    public Object[] getConstructorArgs();
```

Comment:
Great new public method! Thank you for growing the API!

I really like the idea that it automatically includes outer class instance and that it is diligently document here. Nice work.

Let's add @since tag.